api/src/main/java/io/opentelemetry/common/ImmutableKeyValuePairs.java

```diff
@@ -56,6 +57,22 @@ public void forEach(KeyValueConsumer<V> consumer) {
     }
   }
 
+  /**
+   * Returns the value of the given key, or null if the key does not exist.
+   *
+   * <p>Warning: currently implemented via a linear search, so O(n) performance.
```

Comment:
+1 to linear search, these should always be small enough where it's faster and opens up optimizations like deferring / not sorting.

Reply:
I actually implemented a binary search, then backed it out because it seemed like total overkill since we're not expecting many attributes. Good exercise to remind myself of some undergrad CS stuff, though (which I never took, since I don't have a CS degree!).